core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -104,9 +106,10 @@ public String newDataLocation(String filename) {
   }
 
   static class ObjectStoreLocationProvider implements LocationProvider {
-    private static final Function<Object, Integer> HASH_FUNC =
-        Transforms.bucket(Integer.MAX_VALUE).bind(Types.StringType.get());
 
+    private static final HashFunction HASH_FUNC = Hashing.murmur3_32_fixed();
+    private static final BaseEncoding BASE64_ENCODER = BaseEncoding.base64Url().lowerCase().omitPadding();
```

Comment:
Using the Guava encoder allows us to just omit padding. That way we don't have to worry about lengths and get the most bits in the final hash.